Api routes 1/3- refactored my /booking endpt to initialize Booking Service with data from Overview API

hannahmanfredi opened the pull request.

Commit:
Add file for Calibre font

Commit:
Add more css rules

Commit:
Switch CSS libraries: Remove Semantic UI and install styled components

Commit:
Merge branch 'main' of https://github.com/ArrrrWeBeAnneBonny/booking into front-end

Commit:
Change my build script command

Commit:
Fix CORS issue by npm installing cors

Commit:
efactor main /booking endpt so it pings the Overview API on to
initialize my Service with data and handle error

Commit:
Update booking collection's fields to include min_nights

djian001 commented on client/src/Booking.jsx:
```diff
@@ -101,10 +106,31 @@ styleLink.rel = "stylesheet";
 styleLink.href = "https://cdn.jsdelivr.net/npm/semantic-ui/dist/semantic.min.css";
```
this is interesting! why is this added here instead of the HTML?

hannahmanfredi replied:
just another way of doing it!

djian001 commented on client/src/Booking.jsx:
```diff
@@ -101,10 +106,31 @@ styleLink.rel = "stylesheet";
 styleLink.href = "https://cdn.jsdelivr.net/npm/semantic-ui/dist/semantic.min.css";
 document.head.appendChild(styleLink);
 
+class BookingButton extends React.Component {
```
Might want to move this to a new file for cleaner code?

hannahmanfredi replied:
yea it's here for v1 but def will move to its own file, good pt

djian001 commented on package.json:
```diff
@@ -12,19 +12,20 @@
   "scripts": {
     "start": "nodemon ./server/index.js",
-    "react-dev": "webpack --config webpack.config.js",
+    "altStart": "NODE_ENV=production node ./server/index.js",
```
this is interesting, need to look into what this does

hannahmanfredi replied:
that's so i can serve either my proxy or my service!

djian001 commented on server/server.js:
```diff
@@ -1,23 +1,111 @@
+  // add michael's proxy port numb when avail.
+
+app.use(cors());
+app.use(cors({ origin: origins }));
```
from my understanding, if you use app.use(cors()), it allows access to everything so you don't need to double use the app.use(cors({ origin: origins }));

hannahmanfredi replied:
Yup, this is fixed in a diff PR. TY

Commit:
Write mock req to '/booking/book' for testing and npm install moment lib
to get current month/day for inventory

Commit:
My database now stores arrays of arrays of date ranges (including
numbers between start and end of range) which will make it easier for me
to parse on the server side

Commit:
Write API route for '/booking/book' which returns inventory for X months
out
